File: cplusplus/design_pattern/MediatorProj/ConcreteMediator.cpp
#include "stdafx.h"
#include "ConcreteMediator.h"
#include "ConcreteColleague1.h"
#include "ConcreteColleague2.h"


ConcreteMediator::ConcreteMediator()
{
}


ConcreteMediator::~ConcreteMediator()
{
}


void ConcreteMediator::createConcreteMediator()
{
    colleague1 = std::make_shared<ConcreteColleague1>();
    colleague2 = std::make_shared<ConcreteColleague2>();
}

void ConcreteMediator::colleagueChanged(std::shared_ptr<Colleague> clleague)
{
    colleague1->action();
    colleague2->action();
}

File: Java/pattern/MediatorProj/src/com/pattern/ConcreteMediator.java
package com.pattern;

public class ConcreteMediator implements Mediator {
	private ConcreteColleague1 colleague1;
	private ConcreteColleague2 colleague2;
	
	@Override
	public void colleagueChanged(Colleague colleague) {
		colleague1.action();
		colleague2.action();

	}
	
	public void createConcreteMediator() {
		colleague1 = new ConcreteColleague1(this);
		colleague2 = new ConcreteColleague2(this);
	}

}
